src/Builder/Checker/MethodStatementChecker/AbstractMethodStatementChecker.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace SprykerSdk\Integrator\Builder\Checker\MethodStatementChecker;
+
+class AbstractMethodStatementChecker
+{
+    /**
+     * @param mixed $previousValue
+     * @param mixed $currentValue
+     * @param string $field
+     *
+     * @return bool
+     */
+    protected function isExistsStatementsField($previousValue, $currentValue, string $field): bool
```

Comment:
so we have double checking of this . One in the checker class, and one in here

Reply:
Yea because we can't to use recursion execute of `isSameMethodsBody` from `MethodStatementCheckerInterface` classes, it will cause circle dependencies, because in this interface we check only if we have simple structure of statement otherwise we execute recoursion to extract more simple statement.

Reply:
moved to abstract class